units_py/unit_kind.py

```diff
@@ -17,28 +22,146 @@ class UnitKind:
     cost: Optional[Resources]                         = attr.ib()
     font: Optional[Resources]                         = attr.ib(default = None)
 
+
+# TODO: Attack should be None if no attack, not empty
```

Comment:
Disagree. An empty list already has the correct semantics, and `List[int]` is a better type than `Optional[List[int]]` because you don't have to do None checks when using it.

Reply:
+1. As a bit of background, `cost` is `Optional` because we need `None` to represent units that *cannot be bought*, whereas an empty `Resources` means the unit is *free*. `attack` doesn't have the same problem, so using an empty list makes sense and simplifies our code wherever we look at a UnitKind's attack.